spec/automate_liveness_agent/api_client_spec.rb

```diff
@@ -218,4 +229,62 @@
 
   end
 
+  describe "setting up the HTTP Client" do
+    context "default config" do
+      it "uses and verifies SSL" do
+        api_client.load_and_verify_config
+        expect(api_client.http.use_ssl?).to eq(true)
+        expect(api_client.http.verify_mode).to eq(OpenSSL::SSL::VERIFY_PEER)
+      end
+    end
+
+    context "ssl_verify_mode verify_none" do
+      let(:ssl_verify_mode) { "verify_none" }
+
+      it "disables SSL verification" do
+        api_client.load_and_verify_config
+        expect(api_client.http.verify_mode).to eq(OpenSSL::SSL::VERIFY_NONE)
+      end
+    end
+
+    context "invalid ssl mode" do
+      let(:ssl_verify_mode) { "verify_beer" }
+
+      it "raises an error" do
+        expect { api_client.load_and_verify_config }
+          .to raise_error(NameError, /OpenSSL::SSL::VERIFY_BEER/)
+      end
+    end
+
+    context "with a trusted_certs_dir" do
+      let(:trusted_cert_path) { fixture("trusted_certs/chef-example.crt") }
+      let(:trusted_cert) { OpenSSL::X509::Certificate.new(File.read(trusted_cert_path)) }
+      let(:trusted_certs_dir) { File.dirname(trusted_cert_path) }
+
+      it "verifies trusted trusted certs" do
+        api_client.load_and_verify_config
+        expect(api_client.http.cert_store.verify(trusted_cert)).to be_truthy
+      end
+    end
+
+    context "with a ca file" do
+      let(:ssl_ca_file) { "/path/to/custom/ca/file" }
```

Comment:
We should probably also check that the ca cert file exists in the config sanity check phase.